gpt_index/indices/query/query_runner.py

```diff
@@ -74,7 +75,10 @@ def _get_query_kwargs(self, config: QueryConfig) -> Dict[str, Any]:
         return query_kwargs
 
     def query(
-        self, query_str_or_bundle: Union[str, QueryBundle], index_struct: IndexStruct
+        self, 
+        query_str_or_bundle: Union[str, QueryBundle], 
+        index_struct: IndexStruct, 
+        vector_store: Optional[VectorStore] = None,
```

Comment:
yeah as mentioned in the other note, since vector_store is specific to vector store indices (at least atm), it might be nicer to have it be passed through as query_kwargs by calling _preprocess_query on the index class

Reply:
revised.